apps/speed.c

```diff
@@ -417,7 +361,21 @@ const OPTIONS speed_options[] = {
 #define D_IGE_256_AES   28
 #define D_GHASH         29
 #define D_RAND          30
-static OPT_PAIR doit_choices[] = {
+/* name of algorithms to test */
+static const char *names[] = {
+    "md2", "mdc2", "md4", "md5", "hmac(md5)", "sha1", "rmd160", "rc4",
```

Comment:
I wonder what would people think of following pattern
```
static const char *names[] = {
#define D_MD2    0
    "md2",
#define D_MDC2   1
    "mdc2",
...
```
It's still ugly, but at least it gives you right idea about relationship between names and indices....

Reply:
Other suggestion : 
```
enum { D_MD2, D_MDC2, D_MD4,    ...    
static const char *names[] = {
    "md2",    /* D_MD2   */
    "mdc2",     /* D_MDC2   */
    "md4",     /* D_MD4   */
    ...
````

Reply:
Another suggestion (if this is really important) is to have an enum and then a "pair" typedef that has value/string-name.  We do that in many other places.

Reply:
I add these suggestions to the many more ideas I have to simplify and document this code.